src/mpid/ch4/netmod/ofi/ofi_init.c

```diff
@@ -877,6 +878,17 @@ int MPIDI_OFI_post_init(void)
         }
     }
 
+    /* Confirm that all processes have the same number of NICs */
+    MPIR_Errflag_t errflag = MPIR_ERR_NONE;
+    mpi_errno = MPIR_Bcast_allcomm_auto(&num_nics, 1, MPI_INT, 0, MPIR_Process.comm_world,
+                                        &errflag);
+    MPIR_ERR_CHECK(mpi_errno);
+    MPIR_ERR_CHKANDJUMP(num_nics != MPIDI_OFI_global.num_nics, mpi_errno, MPI_ERR_OTHER,
+                        "**ofi_num_nics");
+    /* FIXME: It would also be helpful to check that all of the NICs can communicate so we can fall
+     * back to other options if that is not the case (e.g., verbs are often configured with a
+     * differen subnet for each "set" of nics). It's unknown how to write a good check for that. */
+
```

Comment:
I suggest move this check to the start after line 869. If the `num_nics` is unequal, I think we can just reset the `num_nics` to `1` (and free the extra resources). Since non of the non-zero nic is initialized yet, that should work cleanly.

Reply:
I disagree about always falling back to fewer NICs, however, I'd be open to having a CVAR to enable that behavior. At least in some cases, we want to fail quickly if there is a NIC down.

Reply:
i just added this (along with the CVAR).